src/Process.js

```diff
@@ -58,7 +60,7 @@ Process.prototype.ready = T.func([T.Number], T.Object, "process.ready").of(funct
 
     this.instance.onReady(deferred.resolve);
     this.instance.run();
-    setTimeout(deferred.reject, timeout, new Error("Timeout exceeded."));
+    setTimeout(() => deferred.reject(TimeoutError.create(last(this.instance.instance.output))), timeout);
```

Comment:
ja, du has schon recht. das problem is, dass instance.instance leaky is. bei lastOutput is mein problem, dass es sehr spezifisch is. ich glaube, dass das problem hier daher resultiert, dass die setTimeout funktion auch in die onDeath bzw onReady von instance reingehört. Dann wäre das quasi so iwie:

```js
// Process.js
Process.prototype.death = function(maybeTimeout) {
    if(T.Nil.is(maybeTimeout)) this.instance.kill();

    const timeout  = T.Number(defaults(maybeTimeout).to(0));
    const deferred = Q.defer();

    this.instance.onDeath(deferred);
    this.instance.run();

    return deferred.promise;
};

Process.prototype.ready = T.func([T.Number], T.Object, "process.ready").of(function(timeout) {
    if(this.instance.isRunning() && this.instance.isReady()) return this.instance.lastMatch();

    const deferred = Q.defer();

    this.instance.onReady(deferred);
    this.instance.run();

    return deferred.promise;
});
```

```js
// NodeProcess.js
NodeProcess.prototype.onDeath = T.func([T.Object, T.maybe(Timeout)], T.Nil, "nodeProcess.onDeath").of(function(deferred, timeout = 0) {
    this.emitter.on("death", deferred.resolve);
    this.emitter.on("failure", deferred.reject);

    setTimeout(() => timeout > 0 ? this.onTimeout(deferred) : null, timeout);
});

NodeProcess.prototype.onTimeout = T.func(..).of(function() {
     const error = TimeoutError.create(last(this.instance.output));

     deferred.reject(error);
});

/**
 * Stores a callback, that gets called, when process is ready
 *
 * @param  {Function} cb executed, when process is ready
 */
NodeProcess.prototype.onReady = T.func([T.Object], T.Nil, "nodeProcess.onReady").of(function(deferred) {
    this.emitter.on("ready", deferred.resolve);
    setTimeout(this.onTimeout.bind(this, deferred), timeout);
});
```